src/components/Modals/SetAclPermissionsModal.jsx

```diff
@@ -0,0 +1,213 @@
+
+/**
+ * @typedef {import("../../typedefs.js").setAclPermissionsModalProps} setAclPermissionsModalProps
+ */
+
+/**
+ * SetPermissionsModal Component - Modal component that generates the form for
+ * setting ACL permissions to another user's documents, or document container
+ * in their Solid Pod via Solid Session.
+ *
+ * @memberof Modals
+ * @name SetAclPermissionsModal
+ * @param {setAclPermissionsModalProps} Props - Props for SetAclPermsDocContainerModal component
+ * @returns {React.JSX.Element} The SetAclPermsDocContainerModal Component
+ */
+
+const SetAclPermissionsModal = ({ showModal, setShowModal, dataset }) => {
+  const { session } = useSession();
+  const { state, dispatch } = useStatusNotification();
+  const { podUrl } = useContext(SignedInUserContext);
+  const [docType, setDocType] = useState('');
+  const [permissionState, setPermissionState] = useState({
+    podUrlToSetPermissionsTo: '',
+    permissionType: ''
+  });
+
+  const handleDocType = (event) => {
+    setDocType(event.target.value);
+  };
+
+  const clearInputFields = () => {
+    dispatch({ type: 'CLEAR_PROCESSING' });
+    setPermissionState({
+      podUrlToSetPermissionsTo: '',
+      permissionType: ''
+    });
+    setDocType('');
+    setShowModal(false);
+  };
+
+  // Event handler for setting ACL permissions to file or container on Solid
+  const handleAclPermission = async (event) => {
+    event.preventDefault();
+    dispatch({ type: 'SET_PROCESSING' });
+    const permissions = event.target.setAclPerms.value
+      ? {
+          read: event.target.setAclPerms.value === 'Give',
+          append: event.target.setAclPerms.value === 'Give' && dataset.modalType === 'container'
+        }
+      : undefined;
+    const webIdToSetPermsTo = `${permissionState.podUrlToSetPermissionsTo}profile/card#me`;
+
+    try {
+      switch (dataset.modalType) {
+        case 'container':
+          await setDocContainerAclPermission(session, permissions, podUrl, webIdToSetPermsTo);
+          break;
+        case 'document':
+          await setDocAclPermission(session, docType, permissions, podUrl, webIdToSetPermsTo);
+          break;
+        default:
+          break;
+      }
+
+      runNotification(
+        `${permissions.read ? 'Give' : 'Revoke'} permission to ${
+          permissionState.podUrlToSetPermissionsTo
+        } for ${dataset.modalType === 'container' ? 'Documents Container' : docType}.`,
+        5,
+        state,
+        dispatch
+      );
+    } catch (error) {
+      runNotification('Failed to set permissions. Reason: File not found.', 5, state, dispatch);
+    } finally {
+      setTimeout(() => {
+        clearInputFields();
+      }, 3000);
+    }
+  };
+
+  return (
+    <div>
+      <Dialog open={showModal} onClose={clearInputFields}>
+        <FormSection
+          title={
+            dataset.modalType === 'container'
+              ? 'Permission for all Documents'
+              : 'Permission for Document'
```

Comment:
Just a thought, not sure if it would look good - or be too susceptible of making the modal too wide...

but thought of possibly putting the file name in the title (like `: 'Permission for ${documentName}'` if it's setting permissions for a single document, instead of the `<Typography />` with `File Name: {documentName}`

If that worked/looked good/etc, we could get rid of line 117-131.